config/initializers/cors.rb

```diff
@@ -0,0 +1,14 @@
+Rails.application.config.middleware.insert_before 0, Rack::Cors do
+    allow do
+      origins "http://localhost:5173",
+              "http://127.0.0.1:4000",
+              "http://yourwebsite.production.app",
+              /\Ahttps:\/\/deploy-preview-\d{1,4}--yourwebsite\.domain\.app\z/
+  
+      resource "*",
+        headers: :any,
+        methods: [:get, :post, :put, :patch, :delete, :options, :head],
+        credentials: true,
+        max_age: 86400
+    end
+  end
```

Comment:
- [ ] Nice job using the `rack-cors` gem to solve the CORS policy issue, Unfortunately, You used the wrong configuration, and when I ran the dev server for the react app I had a CORS policy issue in the browser console (see the image below). Please fix that.

![image](https://github.com/ahmed025566/hello-rails-back-end/assets/68796232/c9336b90-ad31-4f4c-b8a8-5c375135a7e2)

```suggestion
Rails.application.config.middleware.insert_before 0, Rack::Cors do
    allow do
      origins '*'
      resource '*', headers: :any, methods: [:get, :post]
    end
end
```